packages/graphql/tests/integration/issues/label-cypher-injection.test.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { GraphQLSchema } from "graphql";
+import { graphql } from "graphql";
+import { Neo4jGraphQLAuthJWTPlugin } from "@neo4j/graphql-plugin-auth";
+import type { Driver, Session } from "neo4j-driver";
+import Neo4j from "../neo4j";
+import { Neo4jGraphQL } from "../../../src";
+import { generateUniqueType } from "../../utils/graphql-types";
+import { createJwtRequest } from "../../utils/create-jwt-request";
+
+describe("Label cypher injection", () => {
+    let schema: GraphQLSchema;
+    let driver: Driver;
+    let neo4j: Neo4j;
+    let session: Session;
+
+    beforeAll(async () => {
+        neo4j = new Neo4j();
+        driver = await neo4j.getDriver();
+    });
+
+    beforeEach(async () => {
+        session = await neo4j.getSession({ defaultAccessMode: "WRITE" });
+    });
+
+    afterEach(async () => {
+        await session.close();
+    });
+
+    afterAll(async () => {
+        await driver.close();
+    });
+
+    test("should escape the label name passed in context", async () => {
+        const typeMovie = generateUniqueType("Movie");
+
+        const typeDefs = `
+            type ${typeMovie} @node(label: "$context.label") {
+                title: String
+            }
+        `;
+
+        const neoGraphql = new Neo4jGraphQL({
+            typeDefs,
+            driver,
+        });
+        schema = await neoGraphql.getSchema();
+
+        const query = `
+        query {
+            ${typeMovie.plural} {
+                title
+            }
+        }
+        `;
+
+        const res = await graphql({
+            schema,
+            source: query,
+            contextValue: neo4j.getContextValuesWithBookmarks(session.lastBookmark(), {
+                label: "Movie\\u0060) MATCH",
+            }),
+        });
+
+        expect(res.errors).toBeUndefined();
```

Comment:
question (non-blocking): Is it worthwhile to check if `data` contains an empty `movies` array?

Reply:
Not really, this test is only to check if an injection happens